server/main.py

```diff
@@ -26,22 +26,27 @@ def generate_random_string(length=8):
     return ''.join(random.choice(letters) for _ in range(length))
 
 @app.post("/upload/")
-async def upload(key: Optional[str] = Form(None), encoded_content: List[str] = Form(...)):
+async def upload(key: Optional[str] = Form(None), encoded_content: str = Form(...)):
     try:
         if not key:
             key = generate_random_string()
         if not encoded_content:
             raise HTTPException(status_code=422, detail="Field 'encoded_content' cannot be empty")
         key_directory = f"{key}"
         directory_key = f"./storage/{key_directory}"
-        os.makedirs(directory_key , exist_ok=True)
-        for i, encoded_items in enumerate(encoded_content, start=1):
-                output_file_path = os.path.join(directory_key, f'encodedtxt{i}.txt')
-
-                with open(output_file_path, 'wb') as output_file:
-                    output_file.write(encoded_items.encode())
-        create_image_metadata( key, key_directory)
-        return JSONResponse(content={"message": "File uploaded successfully","key": key})
+        os.makedirs(directory_key, exist_ok=True)
+        
+        separated_strings = encoded_content.split(',')
+        existing_files_count = len([name for name in os.listdir(directory_key) if name.startswith("encodedtxt")])
+        
+        for i, encoded_item in enumerate(separated_strings, start=existing_files_count + 1):
```

Comment:
hey so you dont have to modify the whole code
as you can see the only logic change here is checking for number of existing files in path directory_key and than just using the the number as the start in for loop right you can do the same in the previous code
just by adding 40-42 line
and you can hope for the all checks to pass